### PR TITLE
Add block count to admin panel

### PR DESCRIPTION
**Pull request by Hamms**
Fixes [#109488326](https://www.pivotaltracker.com/n/projects/1256652/stories/109488326)

Adds counts next to start, recommended, and required blocks in the admin panel.

image

### CODE REVIEW

File: dashboard/app/models/blockly.rb [hunk 1]
```diff
@@ -119,6 +119,21 @@ def pretty_block(block_name)
...
 
+  def self.count_xml_blocks(xml_string)
+    unless xml_string.blank?
+      xml = Nokogiri::XML(xml_string, &:noblanks)
+      # The structure of the XML will be
+      # <document>
+      #   <xml>
+      #     ... blocks ...
+      # So the blocks will be the children of the first child of the document
+      if xml.children && xml.children.first && xml.children.first.children
```
Comment (laurelfan) [inline dashboard/app/models/blockly.rb]: you could one-line this as `return xml.try(:children).try(:first).try(:children).try(:length) || 0`

File: dashboard/app/views/levels/_admin.html.haml [hunk 2]
```diff
@@ -13,9 +13,9 @@
...
                 %li= link_to 'solution', level_edit_blocks_path(@level, :solution_blocks)
               %li= link_to 'toolbox', level_edit_blocks_path(@level, :toolbox_blocks)
-              %li= link_to 'start', level_edit_blocks_path(@level, :start_blocks)
-              %li= link_to 'required', level_edit_blocks_path(@level, :required_blocks)
-              %li= link_to 'recommended', level_edit_blocks_path(@level, :recommended_blocks)
+              %li= link_to "start (#{Blockly.count_xml_blocks(@level.start_blocks)})", level_edit_blocks_path(@level, :start_blocks)
+              %li= link_to "required (#{Blockly.count_xml_blocks(@level.required_blocks)})", level_edit_blocks_path(@level, :required_blocks)
+              %li= link_to "recommended (#{Blockly.count_xml_blocks(@level.recommended_blocks)})", level_edit_blocks_path(@level, :recommended_blocks)
               - if @level.is_a? Artist
                 %li= link_to 'pre-draw', level_edit_blocks_path(@level, :predraw_blocks)
```
Comment (laurelfan) [inline dashboard/app/views/levels/_admin.html.haml:20]: would it be useful to have a count for the predraw, solution, and toolbox blocks also?

File: dashboard/app/models/blockly.rb [hunk 3]
```diff
@@ -119,6 +119,21 @@ def pretty_block(block_name)
     self.class.pretty_print(xml_string)
   end
 
+  def self.count_xml_blocks(xml_string)
```
Comment (laurelfan) [inline dashboard/app/models/blockly.rb:122]: unit test? :)

File: dashboard/app/views/levels/_admin.html.haml (hunk 2 above)
Comment (Hamms) [inline dashboard/app/views/levels/_admin.html.haml:20]: it wasn't requested, but I'll ask em

File: dashboard/app/models/blockly.rb (hunk 1 above)
Comment (Hamms) [inline dashboard/app/models/blockly.rb]: dang that's pretty

File: dashboard/app/models/blockly.rb (hunk 3 above)
Comment (Hamms) [inline dashboard/app/models/blockly.rb:122]: done

File: dashboard/app/views/levels/_admin.html.haml (hunk 2 above)
Comment (Hamms) [inline dashboard/app/views/levels/_admin.html.haml:20]: Asked Kiki, she said they weren't interested.

**Comment by laurelfan** [comment]
LGTM

**Comment by Hamms** [comment]
thanks!

**Pull request merged**